gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -151,8 +160,9 @@ public String toString() {
 
   public DagManager(Config config, boolean instrumentationEnabled) {
     this.config = config;
-    this.queue = new LinkedBlockingDeque<>();
     this.numThreads = ConfigUtils.getInt(config, NUM_THREADS_KEY, DEFAULT_NUM_THREADS);
+    this.queue = initializeDagQueue(this.numThreads);
+    this.cancelQueue = initializeDagQueue(this.numThreads);
```

Comment:
Should we just call the queue controlMessageQueue, to handle future cases where we may want to resume execution of a flow?